Customer tests

Nivirx opened the pull request:
My task was to implement tests for the Customer, Account, and TransactionType's.

Account is a abstract class so it doesn't have any testable methods
TransactionType is a enum and only contains values, so there is nothing to test there either

PR is for my work so far on the Customer JUnit tests (about half done so far).

I would like to start merging in everyone's tests so we can make sure we are all working with the same
versions of the Interfaces because I have had to add small getter/setter methods to complete some of the testing.

__Current testing coverage__

- Test for valid operation of Customer::addAccount()
- Test for valid handling of duplicate accounts added with Customer::addAccount()
- Test for valid return (of null) from Customer::getAccount() if account is non-existant
- Test for valid return from Customer::getAccount if account is existant
- Test that Set<> from getCustomerAccounts is a non-modifiable Set<>
- Test that getCustomerAccounts returns a empty Set<> not null in the event there are no accounts
- Tests that given a set of valid accounts getCustomerAccounts returns a valid set

Nivirx committed:
added tests that make sense for customer

added comments to Account class
added Transaction types to TransactionType enum

Nivirx committed:
About half done with required tests for Customer/Consumer Interface

LlanerasJ commented:
I am still working on it as i could not last night but i will still review and approve it.

Nivirx committed:
renamed test file

basic unit tests to test the required functionality
as outlined on the project guidelines have been completed

All tests should currently fail because nothing is implemented

Nivirx commented:
As of commit cd38a4c all JUnit tests for the Customer class are implemented and failing as normal
